src/MongoDB.Driver.Core/Core/Operations/AsyncCursor.cs

```diff
@@ -270,9 +350,45 @@ protected virtual void Dispose(bool disposing)
                     {
                         _channelSource.Dispose();
                     }
+                    _disposed = true;
+                }
+            }
+        }
+
+        private void CloseCursor(CancellationToken cancellationToken)
```

Comment:
Let's call this method CloseIfNotAlreadyClosed to match the most recent implementation of the pattern in the GridFS class.